### DISCUSSION TITLE
Can I get the real time cost of redisson?

### DISCUSSION THREAD

**Discussion by 24kpure** [apache/skywalking#11799] [Q&A] [answered]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Can I get the execution time of an asynchronous method like 
 `lock` in redisson? In `org.apache.skywalking.apm.plugin.redisson.v3.define.RedissonLockInstrumentation`, it just record the command.Thanks for you answer ~

```
       new InstanceMethodsInterceptPoint() {
                    @Override
                    public ElementMatcher<MethodDescription> getMethodsMatcher() {
                        return named("tryLockInnerAsync").and(takesArgumentWithType(2, "java.util.concurrent.TimeUnit"));
                    }

                    @Override
                    public String getMethodsInterceptor() {
                        return REDISSON_HIGH_LEVEL_LOCK_INTERCEPTOR;
                    }

                    @Override
                    public boolean isOverrideArgs() {
                        return false;
                    }
                }
```

### Are you willing to submit a pull request to implement this on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Async method executes in a short time, that is. That is an async method. Interceptor `REDISSON_HIGH_LEVEL_LOCK_INTERCEPTOR` created span, it is the real-time cost. I am not sure what you are asking.

**Reply by 24kpure** [reply]
First of all,I apologize for not expressing myself well, for my English is not  good. There are a large number of asynchronous methods in Redisson, and  plugin can not get their real-time cost.  Can the official plugin add support for this part？Thanks ~

**Reply by wu-sheng** [reply]
Plugins are not official. Most of the plugins are not built by us. We just accept those. 
The span is able to capture async execution, `asyncStop`. But this has to be added into plugin implementation.
If you want to enhance that plugin, welcome. Some of the Redis plugins are enhanced by others, but may not this(I don't check).

**Reply by 24kpure** [reply]
I get it,it's very nice of you.

**Comment by peachisai** [comment]
@24kpure I will have a try

**Comment by wu-sheng** [comment, marked as answer]
https://github.com/apache/skywalking-java/pull/667 should be able to resolve this.
